### PR TITLE
feat: add  L1CostFunc to evm block context

### PR DESCRIPTION
**Pull request by lightscale-luke**

### CODE REVIEW

**Comment by chokobole** [comment]
1. I think these 2 commits are categorized into `fix`.
2. We decided to follow the [convention](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional). Please don't use `impl`~

**Comment by chokobole** [comment]
Please make this unittest passed.

```
--- FAIL: TestUnderpricingDynamicFee (0.06s)
    txpool_test.go:1881: failed to add well priced transaction: txpool is full
FAIL
FAIL    github.com/ethereum/go-ethereum/core/txpool     13.339s
```

File: eth/tracers/api_blockresult.go [hunk 1]
```diff
@@ -226,6 +227,7 @@ func (api *API) getTxResult(env *traceEnv, state *state.StateDB, index int, bloc
 
 	tracer := vm.NewStructLogger(env.config.LogConfig)
 	// Run the transaction with tracing enabled.
+	env.blockCtx.L1CostFunc = types.NewL1CostFunc(api.backend.ChainConfig(), state)
```
Comment (chokobole) [inline eth/tracers/api_blockresult.go:230]: Considering comment above,  this line should be added above the tracer.

```diff
-tracer := vm.NewStructLogger(env.config.LogConfig)
-// Run the transaction with tracing enabled.
-env.blockCtx.L1CostFunc = types.NewL1CostFunc(api.backend.ChainConfig(), state)
+env.blockCtx.L1CostFunc = types.NewL1CostFunc(api.backend.ChainConfig(), state)
+tracer := vm.NewStructLogger(env.config.LogConfig)
+// Run the transaction with tracing enabled.
```

**Comment by lightscale-luke** [comment]
> Tested by `make test` command and didn't get TestUnderpricingDynamicFee failed as you mentioned. Should I have run tests in different way?

**Comment by chokobole** [comment]
> > Please make this unittest passed.
> > ```
> > --- FAIL: TestUnderpricingDynamicFee (0.06s)
> >     txpool_test.go:1881: failed to add well priced transaction: txpool is full
> > FAIL
> > FAIL    github.com/ethereum/go-ethereum/core/txpool     13.339s
> > ```
> 
> Tested by `make test` command and didn't get `TestUnderpricingDynamicFee` failed as you mentioned. Should I have run tests in different way?

Oh.. I re-ran the test and found it passed. Maybe this is because of the modification to `build/ci.go` locally. FYI, this was the diff to run the tests faster locally.

```diff
diff --git a/build/ci.go b/build/ci.go
index db336a8ef..fd7ec6832 100644
--- a/build/ci.go
+++ b/build/ci.go
@@ -224,7 +224,7 @@ func doInstall(cmdline []string) {
        // better disable it. This check isn't the best, it should probably
        // check for something in env instead.
        if env.CI && runtime.GOARCH == "arm64" {
-               gobuild.Args = append(gobuild.Args, "-p", "1")
+               gobuild.Args = append(gobuild.Args, "-p", strconv.Itoa(runtime.NumCPU()))
        }
 
        // We use -trimpath to avoid leaking local paths into the built executables.
@@ -308,7 +308,7 @@ func doTest(cmdline []string) {
 
        // Test a single package at a time. CI builders are slow
        // and some tests run into timeouts under load.
-       gotest.Args = append(gotest.Args, "-p", "1")
+       gotest.Args = append(gotest.Args, "-p", strconv.Itoa(runtime.NumCPU()))
        if *coverage {
                gotest.Args = append(gotest.Args, "-covermode=atomic", "-cover")
        }
```

**Comment by chokobole** [comment]
Please change the scope of first commit to `eth`

**Review by chokobole — APPROVED** [review]
LGTM

**Pull request merged**